### PR TITLE
Use page result instead of rendering layout directly and add page tit…

### PR DESCRIPTION
**Pull request by mrkhoa99**
### Description

- Use page result instead of rendering layout directly in Advanced Search controller.
- Add Title page on Advanced search page.
### Fixed Issues:

-  Missing Title page on Advanced search page.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=8997) <br/>All committers have signed the CLA.

File: app/code/Magento/CatalogSearch/Controller/Advanced/Result.php [hunk 1]
```diff
@@ -32,33 +25,35 @@ class Result extends \Magento\Framework\App\Action\Action
      *
      * @param Context $context
      * @param ModelAdvanced $catalogSearchAdvanced
-     * @param UrlFactory $urlFactory
```
Comment (ishakhsuvarov) [inline app/code/Magento/CatalogSearch/Controller/Advanced/Result.php]: Due to Backwards Compatibility Policy we can not remove any arguments from the constructor at the moment

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]

File: app/code/Magento/CatalogSearch/Controller/Advanced/Result.php (hunk 1 above)
Comment (mrkhoa99) [reply app/code/Magento/CatalogSearch/Controller/Advanced/Result.php]: @ishakhsuvarov I will fix them. Seem that Travis build failed? Can you give your idea?

**Comment by ishakhsuvarov** [comment]
@mrkhoa99 You can check the [details link](https://travis-ci.org/magento/magento2/jobs/214623508) for the Travis check in this PR. At the moment it failed on unit tests job with the following Fatal Error:
> `PHP Fatal error:  Uncaught Error: Call to a member function getConfig() on null in /home/travis/build/magento/magento2/app/code/Magento/CatalogSearch/Controller/Advanced/Result.php:66`

**Comment by mrkhoa99** [comment]
@ishakhsuvarov seem that we cannot use `Magento\Framework\Controller\ResultFactory` to set the title page in controller? I think it should be ok.

**Pull request closed without merging**